src/game/enemies/kyacchaa.c

```diff
@@ -0,0 +1,468 @@
+    s->x = TO_WORLD_POS_RAW(I(enemy->qPos.x), enemy->region[0]) - gCamera.x;
+    s->y = TO_WORLD_POS_RAW(I(enemy->qPos.y), enemy->region[1]) - gCamera.y - 19;
+
+    // Adjust x position based on sprite's horizontal flip flag
+    if (s->frameFlags & SPRITE_FLAG_MASK_X_FLIP) {
+        s->x -= 1;
+    } else {
+        s->x += 1;
+    }
+
+    acmdRes = UpdateSpriteAnimation(s);
+    DisplaySprite(s);
+
+    // Draw additional sprites vertically above
+    yPos = enemy->qPos.y + Q(7);
+
+    while (yPos > enemy->qUnk18.y) {
+        s = &enemy->s;
+
+        s->x = TO_WORLD_POS_RAW(I(enemy->qPos.x), enemy->region[0]) - gCamera.x;
+        s->y = TO_WORLD_POS_RAW(I(yPos), enemy->region[1]) - gCamera.y - 19;
+
+        // Same x adjustment as the main sprite
+        if (s->frameFlags & SPRITE_FLAG_MASK_X_FLIP) {
+            s->x -= 1;
+        } else {
+            s->x += 1;
+        }
+
+        acmdRes = UpdateSpriteAnimation(s);
+        DisplaySprite(s);
+
+        yPos -= Q(8);
+    }
+
+    return acmdRes;
+}
+
+void Task_Kyacchaa(void)
+{
+    Kyacchaa *enemy = TASK_DATA(gCurTask);
+
+    sub_8065CE0(enemy);
+
+    if (sub_8065B90(enemy) == TRUE) {
+        gCurTask->main = sub_8065A8C;
+        return;
+    }
+
+    if ((gStageData.unk4 != 1) && (gStageData.unk4 != 2) && (gStageData.unk4 != 4)) {
+        sub_8065EB0(enemy);
+    }
+
+    if (sub_8065F5C(enemy) == TRUE) {
+        TaskDestroy(gCurTask);
+    }
+}
+
+void sub_8065E48(void)
+{
+    Kyacchaa *enemy = TASK_DATA(gCurTask);
+    bool32 transformToKyacchaa = FALSE;
+
+    if ((gStageData.unk4 != 1) && (gStageData.unk4 != 2) && (gStageData.unk4 != 4)) {
+        transformToKyacchaa = sub_8065F10(enemy);
+    }
+
+    sub_8065CE0(enemy);
+
+    if (sub_8065F5C(enemy) == TRUE) {
+        TaskDestroy(gCurTask);
+    } else if (transformToKyacchaa == TRUE) {
+        gCurTask->main = Task_Kyacchaa;
+    }
+}
+
+void sub_8065EB0(Kyacchaa *enemy)
+{
+    u16 upperBound = enemy->upperBound;
+    u32 flags = enemy->s2.frameFlags;
+
+    if (flags & SPRITE_FLAG_MASK_X_FLIP) {
+        // Moving upward/right direction
+        if (enemy->qPos.x <= upperBound) {
+            enemy->qPos.x += Q(1);
+
+            if (enemy->qPos.x > upperBound) {
+                enemy->qPos.x = upperBound;
+
+                SPRITE_FLAG_CLEAR(&enemy->s2, X_FLIP);
+                SPRITE_FLAG_CLEAR(&enemy->s, X_FLIP);
+            }
+        }
+    } else {
+        // Moving downward/left direction
+        u16 lowerBound = enemy->lowerBound;
+
+        if (enemy->qPos.x >= lowerBound) {
+            enemy->qPos.x -= Q(1);
+
+            if (enemy->qPos.x < lowerBound) {
+                enemy->qPos.x = lowerBound;
+
+                SPRITE_FLAG_SET(&enemy->s2, X_FLIP);
+                SPRITE_FLAG_SET(&enemy->s, X_FLIP);
+            }
+        }
+    }
+}
+
+bool32 sub_8065F10(Kyacchaa *enemy)
+{
+    if (enemy->qPos.y < enemy->qUnk18.y) {
+        return FALSE;
+    }
+
+    enemy->qPos.y -= 0x80;
```

Comment:
Since it's updating `qPos`, I expected that it could be a `Q` value, but I could not find a suitable Q-value here.